### PR TITLE
WIP: Cabal helper loader

### PR DESCRIPTION
**Pull request by fendor**
Just for show-casing that integrating c-h is not that difficult.

### CODE REVIEW

File: app/Main.hs [hunk 1]
```diff
@@ -14,24 +15,38 @@ import Hie.Locate
...
-  cfs <- runMaybeT $ case name of
-    "cabal" -> cabalPkgs pwd
-    _ -> stackYamlPkgs pwd
-  when (null cfs) $ error $
-    "No .cabal files found under"
-      <> pwd
-      <> "\n You may need to run stack build."
-  pkgs <- catMaybes <$> mapM (nestedPkg pwd) (concat cfs)
-  putStr <$> hieYaml name $ fmtPkgs name pkgs
+  projM <- findCabalHelperEntryPoint $ pwd </> "Foo.hs"
```
Comment (Avi-D-coder) [inline app/Main.hs:25]: This is okay, but should be optional and behind the cabal-helper flag.

File: app/Main.hs [hunk 2]
```diff
@@ -14,24 +15,38 @@ import Hie.Locate
...
+  case projM of
+    Nothing -> error "could not find project context"
+    Just proj@(Ex p) -> do
+      env <- initialiseEnvironment p
+      packages <- loadPackages env
+      pkgs <- toHieYaml env packages
+      let name = if | isCabalProject proj -> "cabal"
```
Comment (Avi-D-coder) [inline app/Main.hs:32]: How does the logic of `isCabalProject` compare to `| any (("dist-newstyle" ==)`?

Comment (fendor) [reply app/Main.hs:32]: That is something to worry about for later. In this case, the project is decided based on the found files, e.g. `cabal.project`, `stack.yaml` and `*.cabal` files. In this order, a project type is selected.

File: app/Main.hs (hunk 1 above)
Comment (fendor) [reply app/Main.hs:25]: True, it is a show-case, it should also not remove the old behaviour as it is waaaaaaaaayyy faster and should be the default mode. So, adding a cli would be sensible, imo, via optparse-applicative

File: app/Main.hs (hunk 2 above)
Comment (Avi-D-coder) [reply app/Main.hs:32]: At some point, we should take build directory into account first so the cabal-helper implicit logic marches the default logic and builds with the users preferred tool.

**Comment by fendor** [comment]
Closed as it seems like we want to go a different way

**Pull request closed without merging**